cloudbuild.yaml

```diff
@@ -49,20 +50,46 @@ steps:
       echo "Get cloud build project configuration values"
       source .env.cloud_build 
 
+      echo "Copy test coverage report from docker volume to a google cloud storage bucket."
       timestamp=$(date +%s)
+      gsutil cp ./coverage/coverage.json "gs://${TEST_COVERAGE_BUCKET_NAME}/${BRANCH_NAME}/${timestamp}-${COMMIT_SHA}.json"
 
-      echo "Copy test coverage report from docker compose volume to Google Gloud Storage."
-      gsutil cp ./coverage/coverage.json "gs://hopic-test-coverage-reports/${BRANCH_NAME}/$timestamp-${COMMIT_SHA}.json"
-
-  - id: "Tear down tests"
-    name: "gcr.io/cloud-builders/docker"
-    script: |
-      #!/usr/bin/env bash
-      set -o errexit
-      set -o pipefail
-      set -o nounset
+      # Install jq and bc 
+      apt-get update && apt-get install -y jq bc
+      
+      # Compare total coverage to failure threshold.
+      total_coverage=$(jq '.totals.percent_covered' ./coverage/coverage.json | awk '{printf "%.2f\n", $1}')
+      if (( $(echo "${total_coverage} < ${TEST_COVERAGE_THRESHOLD}" | bc -l) )); then
+        echo "Coverage (${total_coverage}%) is below the threshold of (${TEST_COVERAGE_THRESHOLD}%) - please add more unit tests!"
+        echo "For more details, read the test coverage report logged in 'Run tests' cloud build step."
+        # Fail build if main branch.
+        if [[ "${BRANCH_NAME}" == 'main' ]]; then  
+          echo "Failing build for insufficent test coverage."  
+          exit 1
+        fi
+      else
+        echo "Coverage (${total_coverage}%) is above the threshold. Continuing..."
```

Comment:
I'd suggest also including the `TEST_COVERAGE_THRESHOLD` value in this output, just for reference.